regression/cbmc/same-object-03/test-no-cp.desc

```diff
@@ -0,0 +1,11 @@
+CORE
+main.c
+--no-simplify --no-propagation
+^EXIT=0$
+^SIGNAL=0$
+^VERIFICATION SUCCESSFUL$
+--
+^warning: ignoring
+--
+Check that __CPROVER_same_object() is false when given two pointers, with one
```

Comment:
Is that expected? malloc is allowed to (and often does) reuse objects

Reply:
I think this depends on how you plan to use `_same_object`? `assert(same_object(p1, p2)` certainly must fail. I think since this is an internal, it should be more exact, in this case, we know they are _not_ the same object, they are the result of two separate mallocs. However, I would expect `assert(p1 != p2)` to fail, since is possible they could have the same address since the memory has been freed.

Reply:
Yes, `assert(p1 != p2)` should fail, though currently the way `malloc` is modelled in cbmc it doesn't reuse addresses.